### PR TITLE
Bugfix - unintended mod wheel values in midi clips

### PR DESCRIPTION
**Pull request by m-m-adams**
Fix issue of sending un needed mod wheel values reported on discord

Fix related issue where some ways of recording mod wheel did not link to mono expression y axis, causing them to be ignored in the MPE collapsing calculation

### CODE REVIEW

File: src/deluge/gui/views/automation_view.cpp [hunk 1]
```diff
@@ -3055,17 +3055,21 @@ void AutomationView::selectNonGlobalParam(int32_t offset, Clip* clip) {
 // used with SelectEncoderAction to get the next midi CC
 void AutomationView::selectMIDICC(int32_t offset, Clip* clip) {
 	if (isOnAutomationOverview()) {
-		clip->lastSelectedParamID = 0;
```
Comment (seangoodvibes) [inline src/deluge/gui/views/automation_view.cpp:3058]: I noticed that when you're on automation overview and scroll right, that it jumps to CC 2 and you have to scroll back to get to CC 0. Can you fix it so the scrolling right when on automation overview brings you to CC 0? Thanks

File: src/deluge/gui/views/automation_view.cpp [hunk 2]
```diff
@@ -3055,17 +3055,21 @@ void AutomationView::selectNonGlobalParam(int32_t offset, Clip* clip) {
...
+		newCC += kNumCCExpression;
 	}
-	else if ((clip->lastSelectedParamID + offset) > kLastMidiCCForAutomation) {
-		clip->lastSelectedParamID = 0;
+	else if (newCC >= kNumCCExpression) {
+		newCC -= kNumCCExpression;
 	}
-	else {
-		clip->lastSelectedParamID += offset;
+	if (newCC == 1) {
```
Comment (seangoodvibes) [inline src/deluge/gui/views/automation_view.cpp:3068]: nit: change 1 to CC_NUMBER_MOD_WHEEL

File: src/deluge/model/clip/instrument_clip.cpp [hunk 3]
```diff
@@ -2927,9 +2927,10 @@ int32_t InstrumentClip::readMIDIParamsFromFile(int32_t readAutomationUpToPos) {
 					else {
 						paramId = stringToInt(contents);
 						if (paramId < kNumRealCCNumbers) {
-							if (paramId == 74) {
-								if (storageManager.firmwareVersionOfFileBeingRead
-								    < FirmwareVersion::FIRMWARE_3P2P0_ALPHA) {
+							if (paramId == 1) {
+								// m-m-adams - used to convert CC74 to y-axis, and I don't think that would
```
Comment (seangoodvibes) [inline src/deluge/model/clip/instrument_clip.cpp:2931]: nit: change 1 to CC_NUMBER_MOD_WHEEL (same below)

File: src/deluge/model/instrument/midi_instrument.cpp [hunk 4]
```diff
@@ -240,11 +227,15 @@ void MIDIInstrument::sendMonophonicExpressionEvent(int32_t whichExpressionDimens
 	}
 
 	case 1: {
-		// mono y expression is limited to positive values, double it to match range
```
Comment (seangoodvibes) [inline src/deluge/model/instrument/midi_instrument.cpp:243]: nit: `case CC_NUMBER_MOD_WHEEL:`

File: src/deluge/model/instrument/midi_instrument.cpp [hunk 5]
```diff
@@ -240,11 +227,15 @@ void MIDIInstrument::sendMonophonicExpressionEvent(int32_t whichExpressionDimens
...
+		// this means that without sending CC1 on the master channel poly expression below 64 will
+		// send as mod wheel 0. However this is better than the alternative of sending erroneous values
+		// because the sound engine initializes MPE-Y as 0 (e.g. a CC value of 64)
+		int32_t polyPart = (lastCombinedPolyExpression[whichExpressionDimension] >> 24);
 		int32_t monoPart = lastMonoExpression[whichExpressionDimension] >> 24;
 		int32_t newValue = std::clamp<int32_t>(polyPart + monoPart, 0, 127);
 		// send CC1 for monophonic expression - monophonic synths won't do anything useful with CC74
+
 		midiEngine.sendCC(masterChannel, 1, newValue, channel);
 		break;
```
Comment (seangoodvibes) [inline src/deluge/model/instrument/midi_instrument.cpp:240]: nit: replace 1 with CC_NUMBER_MOD_WHEEL

File: src/deluge/model/instrument/midi_instrument.cpp [hunk 6]
```diff
@@ -427,6 +418,10 @@ int32_t MIDIInstrument::changeControlNumberForModKnob(int32_t offset, int32_t wh
 	else if (newCC >= kNumCCNumbersIncludingFake) {
 		newCC -= kNumCCNumbersIncludingFake;
 	}
+	if (newCC == 1) {
+		// mod wheel is actually CC_NUMBER_Y_AXIS (122) internally
```
Comment (seangoodvibes) [inline src/deluge/model/instrument/midi_instrument.cpp:422]: nit: change 1 to CC_NUMBER_MOD_WHEEL

File: src/deluge/storage/storage_manager.cpp [hunk 7]
```diff
@@ -1647,14 +1647,17 @@ int32_t StorageManager::readMIDIParamFromFile(int32_t readAutomationUpToPos, MID
...
-			         || !strcmp(contents, "120")) { // We used to write 120 for "none", pre V2.0
+			else if (!strcasecmp(contents, "none")) {
 				cc = CC_NUMBER_NONE;
 			}
 			else {
 				cc = stringToInt(contents);
 			}
-			// TODO: Pre-V2.0 files could still have CC74, so ideally I'd move that to "expression" params here...
+			// will be sent as mod wheel and also map to internal mono expression
+			if (cc == 1) {
```
Comment (seangoodvibes) [inline src/deluge/storage/storage_manager.cpp:1657]: nit: cc == CC_NUMBER_MOD_WHEEL

**Review by seangoodvibes — CHANGES_REQUESTED** [review]

File: src/deluge/model/instrument/midi_instrument.cpp (hunk 4 above)
Comment (m-m-adams) [reply src/deluge/model/instrument/midi_instrument.cpp:243]: In this case it's referring to the expression dimension and not the CC, but I think there's an enum for that too

**Pull request merged**